### ISSUE TITLE
TagMacro: make HKTag summonable for unapplied type lambdas with type bounds

### ISSUE THREAD

**Issue by neko-kai** [zio/izumi-reflect#30]
The following should work:

```scala
trait X
trait XAble[_ <: X]
class Y extends X

def getTag[F[_ <: X]: Tag.auto.T] = Tag[F[Y]]

getTag[XAble]
// Tag[XAble[Y]]
```

**Comment by neko-kai** [comment]
This is already fixed on Scala 3:

```scala
type `TagK<:Dep`[K[_ <: Dep]] = Tag.auto.T[K]

def t[T[_ <: Dep]: `TagK<:Dep`, A <: Dep: Tag] = Tag[T[A]]

assert(t[Trait3, Dep].tag == Tag[Trait3[Dep]].tag)
```

But not fixed on Scala 2.

**Comment by AriajSarkar** [comment]
/attempt #30

# Implementation Plan - Issue 30
## The Issue
So, it looks like `HKTag` fails when we try to summon it for unapplied type lambdas that have type bounds. For instance, if you have `trait X` and `trait XAble[_ <: X]`, trying to get a `Tag[XAble]` either blows up or gives you a tag that ignores the bounds.

## Why it's happening
I dug into `TagMacro.scala`, specifically the `mkHKTagArgStruct` method. The macro tries to grab type parameters from `tpe.typeParams`. The problem is that for these unapplied type lambdas, that list comes up empty during macro expansion, so we lose the type bound info.

## The Fix

### 1. Update TagMacro.scala
I'm going to tweak `mkHKTagArgStruct` in `izumi-reflect/izumi-reflect/src/main/scala-2/izumi/reflect/TagMacro.scala`.

Basically, if `tpe.typeParams` is empty, I'll have it check `tpe.typeSymbol.info` instead. If that turns out to be a `PolyType`, we can grab the parameters from there. This way, we catch those bounds even when the standard list is empty.

### 2. Add a Test Case
I'll add a regression test in `izumi-reflect/izumi-reflect/src/test/scala-2/izumi/reflect/test/Issue30Test.scala`.

It'll just replicate the reported case:
*   Define `trait X` and `trait XAble[_ <: X]`.
*   Try to summon a `Tag` for `XAble`.
*   Make sure it matches what we expect (conceptually `Tag[XAble[Y]]`).

## Verification
*   Run the new test with SBT.
*   Double-check that I haven't broken anything else.
*   https://github.com/AriajSarkar/izumi-reflect/commit/45dc905977e936d5b7f01146d238de683fe2d593
